packages/core/core/src/Environment.js

```diff
@@ -28,65 +29,71 @@ export function createEnvironment({
     }
   }
 
+  let ctx = new Set(typeof context === 'string' ? [context] : context);
+
   if (engines == null) {
-    switch (context) {
-      case 'node':
-      case 'electron-main':
-        engines = {
-          node: DEFAULT_ENGINES.node,
-        };
-        break;
-      case 'browser':
-      case 'web-worker':
-      case 'service-worker':
-      case 'electron-renderer':
-        engines = {
-          browsers: DEFAULT_ENGINES.browsers,
-        };
-        break;
-      default:
-        engines = {};
+    engines = {};
+    for (let context of ctx) {
+      switch (context) {
+        case 'node':
+        case 'electron-main':
+          engines.node = DEFAULT_ENGINES.node;
+          break;
+        case 'browser':
+        case 'web-worker':
+        case 'service-worker':
+        case 'electron-renderer':
+          engines.browsers = DEFAULT_ENGINES.browsers;
+          break;
+      }
     }
   }
 
   if (includeNodeModules == null) {
-    switch (context) {
-      case 'node':
-      case 'electron-main':
-      case 'electron-renderer':
-        includeNodeModules = false;
-        break;
-      case 'browser':
-      case 'web-worker':
-      case 'service-worker':
-      default:
-        includeNodeModules = true;
-        break;
+    includeNodeModules = true;
+    for (let context of ctx) {
+      switch (context) {
+        case 'node':
+        case 'electron-main':
+        case 'electron-renderer':
+          includeNodeModules = false;
+          break;
+        case 'browser':
+        case 'web-worker':
+        case 'service-worker':
+        default:
+          includeNodeModules = true;
+          break;
+      }
```

Comment:
So it depends on the order of the contexts? Maybe actually throw an error if there are conflicting values?